### PR TITLE
Introduce setSources in CompilerStack 

### PR DESCRIPTION
**Pull request by axic**
Part of #3586.

### CODE REVIEW

File: libsolidity/interface/StandardCompiler.cpp [hunk 1]
```diff
@@ -647,8 +647,7 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
 	CompilerStack compilerStack(m_readFile);
 
 	StringMap sourceList = std::move(_inputsAndSettings.sources);
```
Comment (axic) [inline libsolidity/interface/StandardCompiler.cpp:649]: This is used later on too.

**Comment by axic** [comment]
The only test left using `addSource` is `libsolidity/Imports.cpp` which is messy because it calls `addSource` and `compile` repetitively in `name_clash_in_import`, but I think that can be solved by splitting it up into 5 test cases.

Once none of the test cases use `addSource` then it can be removed and with that the flag in `reset` can also be removed.

File: test/libsolidity/SolidityCompiler.cpp [hunk 2]
```diff
@@ -31,7 +31,7 @@ namespace solidity
 namespace test
 {
 
-BOOST_FIXTURE_TEST_SUITE(Compiler, AnalysisFramework)
+BOOST_FIXTURE_TEST_SUITE(SolidityCompiler, AnalysisFramework)
```
Comment (axic) [inline test/libsolidity/SolidityCompiler.cpp]: This is only to remove the annoyance of the test suite not matching the file name.

**Review by chriseth — DISMISSED** [review]

**Review by chriseth — APPROVED** [review]

**Comment by axic** [comment]
Should I squash or it is fine this way?

**Comment by leonardoalt** [comment]
I'd say no need to squash

**Pull request merged**